### PR TITLE
pythonPackages.pytest_xdist: fix build

### PR DESCRIPTION
**Pull request by TredwellGit**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/105368#issuecomment-738433215

Look at https://github.com/NixOS/nixpkgs/commits/staging/pkgs/development/python-modules/pytest-xdist if this is confusing.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
Tested python2Packages.pytest_xdist with https://github.com/NixOS/nixpkgs/pull/75730#issuecomment-566046388
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pytest-xdist/1.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  pname = "pytest-xdist";
+  version = "1.34.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1vh4ps32lp5ignch5adbl3pgchvigdfmrl6qpmhxih54wa1qw3il";
+  };
+
+  nativeBuildInputs = [ setuptools_scm pytest ];
+  checkInputs = [ pytest filelock ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-xdist/1.nix:14]: Can we use pytestCheckHook here?

Comment (TredwellGit) [reply pkgs/development/python-modules/pytest-xdist/1.nix:14]: The diff is deceiving. I did not write this but rather renamed default.nix from before https://github.com/NixOS/nixpkgs/commit/6c1eec9838ab438f5120878ca579db7d2753cdba to 1.nix. This pull request is not about refactoring; it is only about fixing the build.

**Comment by FRidh** [comment]
The relevant changes have landed in staging-next. Please target onto staging-next instead.

File: pkgs/development/python-modules/pytest-xdist/default.nix [hunk 2]
```diff
@@ -1,30 +1,27 @@
-{ stdenv, fetchPypi, buildPythonPackage, execnet, pytest
+{ stdenv, fetchPypi, buildPythonPackage, execnet, pytest_6
```
Comment (FRidh) [inline pkgs/development/python-modules/pytest-xdist/default.nix:1]: but you did make the changes in this file, right?

pytest == pytest 6 so don't change this

File: pkgs/development/python-modules/pytest-xdist/default.nix [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
 
 buildPythonPackage rec {
   pname = "pytest-xdist";
   version = "2.1.0";
+  disabled = !isPy3k;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "82d938f1a24186520e2d9d3a64ef7d9ac7ecdf1a0659e095d18e596b8cbd0672";
+    sha256 = "0wh6pn66nncfs6ay0n863bgyriwsgppn8flx5l7551j1lbqkinc2";
```
Comment (FRidh) [inline pkgs/development/python-modules/pytest-xdist/default.nix:11]: why the hash change?

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pytest-xdist/default.nix (hunk 2 above)
Comment (TredwellGit) [reply pkgs/development/python-modules/pytest-xdist/default.nix:1]: >but you did make the changes in this file, right?

No.

File: pkgs/development/python-modules/pytest-xdist/default.nix (hunk 3 above)
Comment (TredwellGit) [reply pkgs/development/python-modules/pytest-xdist/default.nix:11]: The diff is deceiving. I did not write this but rather renamed 2.nix to default.nix.

File: pkgs/development/python-modules/pytest-xdist/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/pytest-xdist/default.nix:1]: I did this one, pytest_xdist_2 only works with pytest_6. This was back like 2 months ago

**Pull request merged**